core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java

```diff
@@ -952,10 +952,22 @@ public static String getSegmentBlockNameKey(String segID, String blockName,
     String blockNameWithOutPart = blockName
         .substring(blockName.indexOf(CarbonCommonConstants.HYPHEN) + 1,
             blockName.lastIndexOf(CarbonTablePath.getCarbonDataExtension()));
+    // to remove compressor name
+    int index = blockNameWithOutPart.lastIndexOf(CarbonCommonConstants.POINT);
     if (isPartitionTable) {
-      return blockNameWithOutPart;
+      if (index != -1) {
+        return blockNameWithOutPart.replace(blockNameWithOutPart.substring(index), "");
+      } else {
+        return blockNameWithOutPart;
+      }
+    }
+    if (index != -1) {
+      String blockNameWithoutCompressorName =
+          blockNameWithOutPart.replace(blockNameWithOutPart.substring(index), "");
+      return segID + CarbonCommonConstants.FILE_SEPARATOR + blockNameWithoutCompressorName;
+    } else {
+      return segID + CarbonCommonConstants.FILE_SEPARATOR + blockNameWithOutPart;
```

Comment:
Please add a unit test to validate this functionality

Reply:
done